### PR TITLE
[FIX] website: allow to rename a page.

### PR DESCRIPTION
**Pull request by JKE-be**
Big diff for small changes...
Code from delete_page has been splitted (functions, templates, ...)
to be re-usable by rename_page.

### CODE REVIEW

**Comment by jholze** [comment]
@JKE-be First of all, very great Feature, but i got some problems during a test. I created a new page added some content and saved it. Then i tried to changed the name, but the link in the navigation was not updated and also adding new content to the page was not possible, it is getting lost after saving the page.

Regards,

Holze Jens

**Comment by mart-e** [comment]
@jholze for the update links, it's normal, we do not update the existing links but we warn the users about existing links pointing to the page your are renaming.

About the view not being updated, are you sure it's linked to this patch?  Maybe there is a bug in the existing website module (I think there is some issues depending of the clicks you make in a specific order or something, not sure) but I don't think this patch is linked to that.

File: addons/website/models/website.py [hunk 1]
```diff
@@ -224,19 +224,40 @@ def new_page(self, cr, uid, name, template='website.default_page', ispage=True,
...
+        if context is None:
+            context = {}
+        View = self.pool.get('ir.ui.view')
+        view_find = self.key_to_view_id(cr, uid, view_id, context=context)
+        if view_find:
+            v = View.browse(cr, uid, view_find, context=context)
+
+            prefix = v.key.split('.')[0]
+            if not new_name.startswith(prefix):
+                new_name = "%s.%s" % (prefix, new_name)
```
Comment (mart-e) [inline addons/website/models/website.py]: should use `slugify` somewhere

File: addons/website/static/src/js/website.contentMenu.js [hunk 2]
```diff
@@ -70,6 +70,40 @@
...
+                        method: 'rename_page',
+                        args: [
+                            self.mo_id,
+                            val,
+                        ],
+                        kwargs: {
+                            context: context
+                        },
+                    }).then(function () {
+                        window.location = "/page/" + encodeURIComponent(val);
```
Comment (mart-e) [inline addons/website/static/src/js/website.contentMenu.js]: probably need to use the version returned by the controller, after `slugify` call

**Comment by JKE-be** [comment]
Merged

**Pull request closed without merging**